docs/architecture/adr-015-namespace-id-size.md

```diff
@@ -0,0 +1,212 @@
+20                        | 20 + 512 = 532             | 2*20 + 32 = 72
+32                        | 32 + 512 = 544             | 2*32 + 32 = 96
+
+### NMT Proof Size
+
+Increasing the size of NMT nodes will increase the size of the NMT proof. Assuming shares are 512 bytes, square size is 128, and the NMT proof is for a single leaf:
+
+Namespace ID size (bytes) | Unencoded NMT proof size (bytes) | Protobuf encoded NMT proof size (bytes) | Protobuf encoded NMT proof with [gzip](https://pkg.go.dev/compress/gzip) (bytes)
+--------------------------|----------------------------------|-----------------------------------------|---------------------------------------------------------------------------------
+8                         | 336                              | 354                                     | 382
+16                        | 448                              | 466                                     | 408
+20                        | 504                              | 522                                     | 466
+32                        | 672                              | 690                                     | 630
+
+### Blob Inclusion Proof Size
+
+Blob inclusion proofs haven't yet been implemented so this proposal can't precisely determine the impact on blob inclusion proofs. A naive implementation of blob inclusion proofs may return NMT proofs for all shares that a blob occupies, in other words one NMT proof per row that a blob spans. Assuming shares are 512 bytes, square size is 128, and a blob is less than 128 shares, a blob would occupy a maximum of 2 rows. Therefore, the namespace ID size's impact on blob inclusion proofs would be approximately 2 * the impact on NMT proofs. A [blob size independent inclusion proof](https://github.com/celestiaorg/celestia-app/blob/6d27b78aa64a749a808e84ea682352b8b551fbd7/docs/architecture/adr-011-optimistic-blob-size-independent-inclusion-proofs-and-pfb-fraud-proofs.md?plain=1#L19) is likely smaller than this naive implementation because it depends on the number of shares that a PFB transaction spans (likely significantly fewer than 2 rows).
+
+### Share Size
+
+Another tradeoff to consider is the size of the namespace ID in the share. Since a share is a fixed 512 bytes, a share's capacity for blob data decreases as the namespace ID increases.
+
+| Namespace ID size (bytes) | Namespace ID size (bytes) / 512 (bytes) |
+|---------------------------|-----------------------------------------|
+| 8                         | 1.5%                                    |
+| 16                        | 3.1%                                    |
+| 20                        | 3.9%                                    |
+| 32                        | 6.2%                                    |
+
+## Open Questions
+
+1. What are the performance implications on celestia-node for a larger namespace ID size?
+1. Is it possible to mitigate some tradeoffs when adopting a large namespace ID size?
```

Comment:
[nit] All the items have been assigned the same number "1", it needs to be fixed.

Reply:
resolving b/c https://www.markdownguide.org/basic-syntax/#ordered-lists